src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1843,6 +1843,72 @@ impl Cage {
         }
     }
 
+    //------------------------------------FSYNC SYSCALL------------------------------------
+
+    pub fn fsync_syscall(&self, fd: i32) -> i32 {
+        let checkedfd = self.get_filedescriptor(fd).unwrap();
+        let mut unlocked_fd = checkedfd.write();
+        if let Some(filedesc_enum) = &mut *unlocked_fd {
+            match filedesc_enum {
+                File(ref mut normalfile_filedesc_obj) => {
+                    if is_rdonly(normalfile_filedesc_obj.flags) {
+                        return syscall_error(Errno::EBADF, "fsync", "specified file not open for sync");
```

Comment:
Can we not sync read only files or are we choosing not to sync them?

Reply:
nothing can be written to them so they shouldnt be synced